monai/apps/auto3dseg/data_analyzer.py

```diff
@@ -235,22 +316,29 @@ def get_all_case_stats(self, key="training", transform_list=None):
             collate_fn=no_collation,
             pin_memory=self.device.type == "cuda",
         )
-        result: dict[DataStatsKeys, Any] = {DataStatsKeys.SUMMARY: {}, DataStatsKeys.BY_CASE: []}
         result_bycase: dict[DataStatsKeys, Any] = {DataStatsKeys.SUMMARY: {}, DataStatsKeys.BY_CASE: []}
-
+        device = self.device if self.device.type == "cpu" else torch.device("cuda", rank)
         if not has_tqdm:
             warnings.warn("tqdm is not installed. not displaying the caching progress.")
 
-        for batch_data in tqdm(dataloader) if has_tqdm else dataloader:
+        for batch_data in tqdm(dataloader) if (has_tqdm and rank == 0) else dataloader:
             batch_data = batch_data[0]
-            batch_data[self.image_key] = batch_data[self.image_key].to(self.device)
-
-            if self.label_key is not None:
-                label = batch_data[self.label_key]
-                label = torch.argmax(label, dim=0) if label.shape[0] > 1 else label[0]
-                batch_data[self.label_key] = label.to(self.device)
-
-            d = summarizer(batch_data)
+            try:
+                batch_data[self.image_key] = batch_data[self.image_key].to(device)
+                if self.label_key is not None:
+                    label = batch_data[self.label_key]
+                    label = torch.argmax(label, dim=0) if label.shape[0] > 1 else label[0]
+                    batch_data[self.label_key] = label.to(device)
+
+                d = summarizer(batch_data)
+            except Exception:
+                logger.info(f"Unable to process data {batch_data['image_meta_dict']['filename_or_obj']} on {device}.")
+                if self.device.type == "cuda":
+                    logger.info("Data analysis using CPU.")
```

Comment:
Data analysis is using CPU, or GPU?

Reply:
The info may be improved here to avoid confusion: 
```
DataAnalyzer `device` was set to use GPU but failed. Falling back to use `cpu`.
```

Reply:
Added this log info. Thanks.